### PR TITLE
MAINT Split large TS functions add add unit tests

### PR DESCRIPTION
**Pull request by ryanking13**
### Description

As title. Split large TS functions, mostly in load-package.ts, into smaller ones and add unit tests for those functions to improve TS test coverage.

### Checklists

- [x] Add / update tests

### CODE REVIEW

File: src/js/load-package.ts [hunk 1]
```diff
@@ -173,46 +182,40 @@ export class PackageManager {
...
+      const loadedChannel = this.getLoadedPackage(name);
+      if (!loadedChannel) continue;
 
       toLoad.delete(normalizedName);
-      // If uri is from the DEFAULT_CHANNEL, we assume it was added as a
+      // If uri is from the default channel, we assume it was added as a
       // dependency, which was previously overridden.
-      if (loaded === channel || channel === DEFAULT_CHANNEL) {
-        messageCallback(`${name} already loaded from ${loaded}`);
+<<<<<<< HEAD
```
Comment (hoodmane) [inline src/js/load-package.ts:197]: Merge conflict marker

Comment (ryanking13) [reply src/js/load-package.ts]: Oops.

File: src/js/load-package.ts [hunk 2]
```diff
@@ -541,6 +554,29 @@ function filterPackageData({
   return { name, version, fileName: file_name, packageType: package_type };
 }
 
+/**
+ * Converts a string or PyProxy to an array of strings.
+ * @private
+ */
+export function toStringArray(
+  str: string | PyProxy | Array<string>,
+): Array<string> {
```
Comment (hoodmane) [inline src/js/load-package.ts:563]:
```suggestion
export function toStringArray(
  str: string | PyProxy | string[],
): string[] {
```

File: src/js/load-package.ts [hunk 3]
```diff
@@ -530,6 +521,28 @@ export class PackageManager {
   public setCdnUrl(url: string) {
     this.cdnURL = url;
   }
+
+  /**
+   * getLoadedPackage returns the channel from which a package was loaded.
+   * if the package is not loaded, it returns null.
+   * @param pkg package name
+   */
+  public getLoadedPackage(pkg: string): string | null {
```
Comment (hoodmane) [inline src/js/load-package.ts:530]: Maybe the name should include channel?

File: src/js/test/unit/package-manager.test.ts [hunk 4]
```diff
@@ -1,59 +1,93 @@
...
+    chai.assert.deepEqual(toStringArray(["hello", "world"]), [
+      "hello",
+      "world",
+    ]);
+  });
+
+  it("Should convert PyProxy to array of strings", () => {
+    // TODO: use real PyProxy
+    const pyProxyMock = {
+      toJs: () => "hello",
```
Comment (hoodmane) [inline src/js/test/unit/package-manager.test.ts:70]: While you're at it, maybe add another where `toJs()` returns a list? This is the more realistic case, I don't know if `toJs` would ever return a string in real life.

**Review by hoodmane — APPROVED** [review]
Thanks!

File: src/js/test/unit/package-manager.test.ts (hunk 4 above)
Comment (ryanking13) [reply src/js/test/unit/package-manager.test.ts]: I changed the test to return the list. Thanks for the suggestion.

File: src/js/load-package.ts (hunk 3 above)
Comment (ryanking13) [reply src/js/load-package.ts]: Changed to `getLoadedPackageChannel`.

**Pull request merged**